aten/src/ATen/native/ReplicationPadding.cpp

```diff
@@ -211,12 +210,621 @@ Tensor& replication_pad1d_backward_out_cpu_template(
+        if (j < pad_l) {
+          ip_x = pad_l;
+        } else if (j >= pad_l && j < iwidth + pad_l) {
+          ip_x = j;
+        } else {
+          ip_x = iwidth + pad_l - 1;
+        }
+        ip_x = ip_x - oStartX + iStartX;
+
+        if (i < pad_t) {
+          ip_y = pad_t;
+        } else if (i >= pad_t && i < iheight + pad_t) {
+          ip_y = i;
+        } else {
+          ip_y = iheight + pad_t - 1;
+        }
+        ip_y = ip_y - oStartY + iStartY;
+
+        scalar_t *src_p = goutput_p + k*owidth*oheight + i * owidth + j;
+        scalar_t *dest_p = ginput_p + k*iwidth*iheight + ip_y * iwidth + ip_x;
+        *dest_p += *src_p;
+      }
+    }
+  }
+}
+
+Tensor& replication_pad2d_backward_out_cpu_template(
+    Tensor& gradInput,
+    const Tensor& gradOutput_,
+    const Tensor& input,
+    IntList paddingSize)
+{
+  int pad_l = paddingSize[0];
+  int pad_r = paddingSize[1];
+  int pad_t = paddingSize[2];
+  int pad_b = paddingSize[3];
+  int dimw = 2;
+  int dimh = 1;
+  int dimslices = 0;
+  int64_t nbatch = 1;
+  int64_t nslices;
+  int64_t iheight;
+  int64_t iwidth;
+  int64_t oheight;
+  int64_t owidth;
+
+  if (input.dim() == 4)
+  {
+    nbatch = input.size(0);
+    dimw++;
+    dimh++;
+    dimslices++;
+  }
+
+  /* sizes */
+  nslices = input.size(dimslices);
+  iheight = input.size(dimh);
+  iwidth = input.size(dimw);
+  oheight = iheight + pad_t + pad_b;
+  owidth  = iwidth + pad_l + pad_r;
+
+  AT_CHECK(owidth == gradOutput_.size(dimw),
+      "gradOutput width unexpected. Expected: ", owidth, ", Got: ",
+      gradOutput_.size(dimw));
+  AT_CHECK(oheight == gradOutput_.size(dimh),
+      "gradOutput height unexpected. Expected: ", oheight, ", Got: ",
+      gradOutput_.size(dimh));
+
+  /* get contiguous gradOutput */
+  auto gradOutput = gradOutput_.contiguous();
+
+  /* resize */
+  gradInput.resize_as_(input);
+  gradInput.zero_();
+
+  /* backprop */
+  if (input.dim() == 3) {
+    AT_DISPATCH_FLOATING_TYPES(
+        input.type(), "replication_pad2d_backward", [&] {
+        replication_pad2d_backward_out_frame<scalar_t>(
+            gradInput.data<scalar_t>(),
+            gradOutput.data<scalar_t>(),
+            nslices,
+            iwidth, iheight,
+            owidth, oheight,
+            pad_l, pad_r,
+            pad_t, pad_b);
+        }
+        );
+  } else {
+    int64_t p;
+#pragma omp parallel for private(p)
+    for (p = 0; p < nbatch; p++) {
+      AT_DISPATCH_FLOATING_TYPES(
+          input.type(), "replication_pad2d_backward", [&] {
+          replication_pad2d_backward_out_frame<scalar_t>(
+              gradInput.data<scalar_t>() + p * nslices * iheight * iwidth,
+              gradOutput.data<scalar_t>() + p * nslices * oheight * owidth,
+              nslices,
+              iwidth, iheight,
+              owidth, oheight,
+              pad_l, pad_r,
+              pad_t, pad_b);
+          }
+          );
+    }
+  }
+  return gradInput;
+}
+
+static inline void shapeCheck(
```

Comment:
It would be good to mention in the name that this function is specifically for 3d only.